rig/machine_control/machine_controller.py

```diff
@@ -1378,6 +1388,17 @@ def __str__(self):
             "Failed to load applications to cores {}".format(", ".join(cores)))
 
 
+def _if_not_closed(f):
+    """Run the method iff. the memory view hasn't been closed."""
+    @functools.wraps(f)
```

Comment:
Sadly, you may well need [`add_signature_to_docstring`](https://github.com/project-rig/rig/blob/master/rig/utils/docstrings.py#L51) to make the docs come out correctly after this :(. Sigh... Python 2...